src/AccessToken.php

```diff
@@ -0,0 +1,308 @@
+use Google\Auth\HttpHandler\HttpClientCache;
+use Google\Auth\HttpHandler\HttpHandlerFactory;
+use GuzzleHttp\Psr7;
+use GuzzleHttp\Psr7\Request;
+use phpseclib\Crypt\RSA;
+use phpseclib\Math\BigInteger;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * Wrapper around Google Access Tokens which provides convenience functions
+ */
+class AccessToken
+{
+    const FEDERATED_SIGNON_CERT_URL = 'https://www.googleapis.com/oauth2/v3/certs';
+    const OAUTH2_ISSUER = 'accounts.google.com';
+    const OAUTH2_ISSUER_HTTPS = 'https://accounts.google.com';
+    const OAUTH2_REVOKE_URI = 'https://oauth2.googleapis.com/revoke';
+
+    /**
+     * @var callable
+     */
+    private $httpHandler;
+
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cache;
+
+    /**
+     * @param callable $httpHandler [optional] An HTTP Handler to deliver PSR-7 requests.
+     * @param CacheItemPoolInterface $cache [optional] A PSR-6 compatible cache implementation.
+     */
+    public function __construct(
+        callable $httpHandler = null,
+        CacheItemPoolInterface $cache = null
+    ) {
+        // @codeCoverageIgnoreStart
+        if (!class_exists('phpseclib\Crypt\RSA')) {
+            throw new \RuntimeException('Please require phpseclib/phpseclib v2 to use this utility.');
+        }
+        // @codeCoverageIgnoreEnd
+
+        $this->httpHandler = $httpHandler
+            ?: HttpHandlerFactory::build(HttpClientCache::getHttpClient());
+        $this->cache = $cache ?: new MemoryCacheItemPool();
+        $this->configureJwtService();
+
+        // set phpseclib constants if applicable
+        $this->setPhpsecConstants();
+    }
+
+    /**
+     * Verifies an id token and returns the authenticated apiLoginTicket.
+     * Throws an exception if the id token is not valid.
+     * The audience parameter can be used to control which id tokens are
+     * accepted.  By default, the id token must have been issued to this OAuth2 client.
+     *
+     * @param string $token The JSON Web Token to be verified.
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type string $audience The indended recipient of the token.
+     *     @type string $certsLocation The location (remote or local) from which
+     *        to retrieve certificates, if not cached. This value should only be
+     *        provided in limited circumstances in which you are sure of the
+     *        behavior.
+     * }
+     * @return array|bool the token payload, if successful, or false if not.
+     */
+    public function verify($token, array $options = [])
+    {
+        $options += [
+            'audience' => null,
+            'certsLocation' => self::FEDERATED_SIGNON_CERT_URL,
+        ];
+
+        // Check signature against each available cert.
+        // allow the loop to complete unless a known bad result is encountered.
+        $certs = $this->getFederatedSignOnCerts($options['certsLocation'], $options);
```

Comment:
It looks like we need to make sure audience/certsLocation get pulled from `$options` before passing it through here.